.github/workflows/build_and_test.yml

```diff
@@ -103,54 +117,115 @@ jobs:
         run: |
           cat /proc/cpuinfo
 
-      - name: Download build-native-${{ matrix.compiler }} artifacts
+      - name: Download build-x86_64-${{ matrix.compiler }}-${{ matrix.compiler_version }} artifacts
         uses: actions/download-artifact@v3
         with:
-          name: build-native-${{ matrix.compiler }}
+          name: build-x86_64-${{ matrix.compiler }}-${{ matrix.compiler_version }}
 
-      - name: Fix _build-native permissions
+      - name: Fix _build-x86_64 permissions
         run: |
-          chmod +x _build-native/bin/*
+          chmod +x _build-x86_64/bin/*
 
-      - name: Test native
+      - name: Test x86_64
         env:
           CTEST_OUTPUT_ON_FAILURE: "TRUE"
         run: |
-          cd _build-native
+          cd _build-x86_64
           ctest -j$(nproc)
 
-      - name: Upload test-native artifacts
+      - name: Upload test-x86_64 artifacts
         uses: actions/upload-artifact@v3
         with:
-          name: test-native
+          name: test-x86_64
           path: |
-            _build-native/Testing
+            _build-x86_64/Testing
         if: always()
 
   build-cross:
     runs-on: ubuntu-latest
-    needs: [build-native]
+    needs: [build-x86_64]
     strategy:
       fail-fast: false
       matrix:
-        arch: [aarch64, armhf, ppc64el, s390x, riscv64]
-        compiler: [gcc, llvm]
```

Comment:
It's a shame to lose this to a very long list that is difficult to read/maintain, and it gets worse for test-cross.
This looks quite drastic for simply changing the compiler version.

Does it make it easier to include the versions in the `compiler` variable directly and keep a structure similar to the previous one, that seemed more contained?

It might not be very idiomatic but can we filter non-functional compilers and versions in a different way that doesn't involve (just) includes and excludes, maybe with basic conditionals and additional steps?

If not is there any better way to display this list of includes/excludes?